net/net.go

```diff
@@ -127,48 +131,71 @@ func (t *net) Close() (err error) {
 		}
 	}
 
-	weakClose("DAGService", t.DAGService)
-	weakClose("host", t.host)
-	weakClose("threadstore", t.store)
+	weakClose("DAGService", n.DAGService)
+	weakClose("host", n.host)
+	weakClose("threadstore", n.store)
 
-	t.bus.Discard()
-	t.cancel()
+	n.bus.Discard()
+	n.cancel()
 
 	if len(errs) > 0 {
 		return fmt.Errorf("failed while closing net; err(s): %q", errs)
 	}
 
-	t.pullLock.Lock()
-	defer t.pullLock.Unlock()
+	n.pullLock.Lock()
+	defer n.pullLock.Unlock()
 	// Wait for all thread pulls to finish
-	for _, semaph := range t.pullLocks {
+	for _, semaph := range n.pullLocks {
 		semaph <- struct{}{}
 	}
 	return nil
 }
 
-func (t *net) Host() host.Host {
-	return t.host
+func (n *net) Host() host.Host {
+	return n.host
 }
 
-func (t *net) Store() lstore.Logstore {
-	return t.store
+func (n *net) Store() lstore.Logstore {
+	return n.store
 }
 
-func (t *net) GetHostID(_ context.Context) (peer.ID, error) {
-	return t.host.ID(), nil
+func (n *net) GetHostID(_ context.Context) (peer.ID, error) {
+	return n.host.ID(), nil
 }
 
-func (t *net) CreateThread(_ context.Context, creds thread.Credentials, opts ...core.KeyOption) (info thread.Info, err error) {
-	id := creds.ThreadID()
-	if err = t.ensureUnique(id); err != nil {
+func (n *net) GetToken(_ context.Context, identity thread.Identity) (tok thread.Token, err error) {
+	msg := make([]byte, tokenChallengeBytes)
+	if _, err = rand.Read(msg); err != nil {
+		return
+	}
+	sig, err := identity.Sign(msg)
+	if err != nil {
 		return
 	}
+	key := identity.GetPublic()
+	if ok, err := key.Verify(msg, sig); !ok || err != nil {
+		return tok, fmt.Errorf("bad signature")
+	}
+	return thread.NewToken(n.getPrivKey(), key)
```

Comment:
Finally, return a newly minted token, signed by the host's private key. This is saying, "Okay, as far as I'm concerned, you are who you say you are, and you can use this verifiable token in the future to remind me".